File: Ejercicios C/13.Tabla de la verdad.cpp
#include <iostream>
using namespace std;

int main() {
    // Variables booleanas
    bool a = true;
    bool b = false;
    
    // Operador AND (&&)
    cout << "Resultado de a && b: " << (a && b) << endl;
    // El resultado será "false" porque b es false.

    // Variables numéricas
    int x = 2, y = 3, z = 4, w = 10;

    // Operador AND (&&) con comparaciones
    cout << "Resultado de (x == y && z < w): " << ((x == y) && (z < w)) << endl;
    // Resultado: (x == y) es false, por lo tanto, el operador && devuelve "false".

    // Operador NOT (!) en una expresión
    cout << "Resultado de !a == (y && z > w): " << (!(a == (y && z > w))) << endl;
    // El operador NOT invierte el valor de la comparación de "a" con el resultado de la comparación (y && z > w).
    // Aquí, (y && z > w) es false, entonces el operador "!" invierte el valor de "true".

    return 0;
}

File: Ejercicios PHP/13.Tabla de la verdad.php
<?php
// Tabla de la verdad

// El operador lógico && solo devuelve true si ambos valores son true.
$a = true;
$b = false;
echo($a && $b) ? 'true' : 'false';// El resultado será "false" porque b es igual a false
echo("\n"); 
// Comparación de $a y $b, y luego $c < $d con operador lógico &&
$a = 2;
$b = 3;
$c = 4;
$d = 10;
//? 'true' : 'false'; en php true es = 1 y false no imprime nada, por esta razon agregamos true y false como cadena de texto
echo ($a == $b && $c < $d) ? 'true' : 'false'; // El resultado será "false" porque $a no es igual a $b
echo "\n"; // Salto de línea para mejorar la legibilidad

// El operador lógico ! invierte el valor de la expresión booleana
// Se evalúa primero la expresión ($a == $b), se invierte con ! y luego se compara con ($c > $d)
echo(!($a == $b) && ($c > $d) ) ? 'true' : 'false'; // El resultado será "true" porque ($a == $b) es false, y ! lo convierte en true.

// Para ejecutar en terminal:
// php "C:\Sena\Analisis y desarrollo de software\Ficha 3066478 segundo trimestre\Ejercicios PHP\13.Tabla de la verdad.php"
?>
